google/cloud/bigquery/v2/minimal/internal/json_utils.cc

```diff
@@ -93,6 +93,20 @@ void ToJson(std::chrono::system_clock::time_point const& field,
   j[name] = std::to_string(m);
 }
 
+nlohmann::json RemoveEmptyArraysAndObjects(nlohmann::json j) {
+  for (auto item = j.begin(); item != j.end(); ) {
+    if ((item.value().is_array() || item.value().is_object()) && !item.value().empty()) {
+      item.value() = RemoveEmptyArraysAndObjects(std::move(*item));
```

Comment:
Tying the client call-stack depth to the nesting depth of user-supplied data sounds problematic.  Can we use a data-space stack instead?